Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/ConvertRebar.cs

```diff
@@ -1,175 +1,332 @@
+              .ToList()
+          );
+        }
+      }
 
-      for (int i = 0; i < bars.Count; i++)
+      // get plane normal of rebar group
+      // the normal prop was deprecated in revit 2018, and the accessor normal is suggested as a replacement
+      // unclear how non-shape-driven rebar set normals are retrieved or computed.
+      Vector normal = null;
+      if (accessor != null)
       {
-        var bar = (accessor != null) ? bars[i].CreateTransformed(accessor.GetBarPositionTransform(i)) : bars[i];
-        curves.Add(CurveToSpeckle(bar, revitRebar.Document));
+        normal = VectorToSpeckle(accessor.Normal, revitRebar.Document);
       }
 
-      var speckleRebar = new RevitRebar();
-      speckleRebar.host = revitRebar.GetHostId().ToString();
-      speckleRebar.type = revitRebar.Document.GetElement(revitRebar.GetTypeId()).Name;
-      speckleRebar.curves = curves;
-      speckleRebar.shapes = revitRebar.GetAllRebarShapeIds().Select(o => o.ToString()).ToList(); // freeform rebar with bent workshop has multiple shapes
+      // create speckle rebar
+      var speckleRebar = new RevitRebarGroup();
+      speckleRebar.shape = speckleShape;
+      speckleRebar.startHook = speckleStartHook;
+      speckleRebar.endHook = speckleEndHook;
+      speckleRebar.number = revitRebar.Quantity;
+      speckleRebar.hasFirstBar = isSingleLayout ? true : revitRebar.IncludeFirstBar;
+      speckleRebar.hasLastBar = isSingleLayout ? true : revitRebar.IncludeLastBar;
       speckleRebar.volume = revitRebar.Volume;
+      speckleRebar.family = type?.FamilyName;
+      speckleRebar.type = type?.Name;
+      speckleRebar.normal = normal;
+      speckleRebar.barPositions = revitRebar.NumberOfBarPositions;
+      speckleRebar.displayValue = centerlines;
 
+      // skip display value as meshes for now
+      // GetElementDisplayValue(revitRebar, SolidDisplayValueOptions);
       GetAllRevitParamsAndIds(speckleRebar, revitRebar);
-
       return speckleRebar;
     }
 
+    // rebar shape
+    private RevitRebarShape RebarShapeToSpeckle(DB.Structure.RebarShape revitRebarShape)
+    {
+      // get the type of the shape
+      RebarType rebarType = RebarType.Unknown;
+      switch (revitRebarShape.RebarStyle)
+      {
+        case RebarStyle.Standard:
+          rebarType = RebarType.Standard;
+          break;
+        case RebarStyle.StirrupTie:
+          rebarType = RebarType.StirrupPolygonal;
+          break;
+      }
+
+      // get the curves representing the default values of the shape
+      List<ICurve> curves = revitRebarShape
+        .GetCurvesForBrowser()
+        .Select(o => CurveToSpeckle(o, revitRebarShape.Document))
+        .ToList();
+
+      var speckleRebarShape = new RevitRebarShape();
+      speckleRebarShape.name = revitRebarShape.Name;
+      speckleRebarShape.curves = curves;
+      speckleRebarShape.rebarType = rebarType;
+
+      GetAllRevitParamsAndIds(speckleRebarShape, revitRebarShape);
+      return speckleRebarShape;
```

Comment:
I might be include to add a line of WS before the return so it can be a little easier to see